internal/lang/funcs/encoding.go

```diff
@@ -18,22 +18,24 @@ import (
 var Base64DecodeFunc = function.New(&function.Spec{
 	Params: []function.Parameter{
 		{
-			Name: "str",
-			Type: cty.String,
+			Name:        "str",
+			Type:        cty.String,
+			AllowMarked: true,
 		},
 	},
 	Type: function.StaticReturnType(cty.String),
 	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
-		s := args[0].AsString()
+		str, strMarks := args[0].Unmark()
+		s := str.AsString()
 		sDec, err := base64.StdEncoding.DecodeString(s)
 		if err != nil {
-			return cty.UnknownVal(cty.String), fmt.Errorf("failed to decode base64 data '%s'", s)
+			return cty.UnknownVal(cty.String).WithMarks(strMarks), fmt.Errorf("failed to decode base64 data %s", redactIfSensitive(s, strMarks))
 		}
 		if !utf8.Valid([]byte(sDec)) {
-			log.Printf("[DEBUG] the result of decoding the provided string is not valid UTF-8: %s", sDec)
-			return cty.UnknownVal(cty.String), fmt.Errorf("the result of decoding the provided string is not valid UTF-8")
+			log.Printf("[DEBUG] the result of decoding the provided string is not valid UTF-8: %s", redactIfSensitive(sDec, strMarks))
+			return cty.UnknownVal(cty.String).WithMarks(strMarks), fmt.Errorf("the result of decoding the provided string is not valid UTF-8")
```

Comment:
Do we have a precedent for returning the arguments' marks on an unknown value with an error? The unknown value from the error shouldn't need to propagate the marks any further, and I don't think we're returning marks in all cases below.

Reply:
Oh, good point! This was the precedent I cargo-culted: https://github.com/hashicorp/terraform/blob/ba6a64eb35acccf1c2775eed81652489b71f4e30/internal/lang/funcs/collection.go#L314-L315

But having poked around inside HCL for a bit I'm really pretty sure that the value is unused if we return an error from a function. I'll remove the marks.

Reply:
Done!